src/shared/dm-util.c

```diff
@@ -5,3 +5,43 @@
 #include "dm-util.h"
 #include "fd-util.h"
 #include "string-util.h"
+
+int dm_deferred_remove_cancel(const char *name) {
+        _cleanup_close_ int fd = -1;
+        size_t dm_size = sizeof(struct dm_ioctl) + sizeof(struct dm_target_msg) + STRLEN("@cancel_deferred_remove") + 1;
+        struct {
+                struct dm_ioctl dm_ioctl;
+                struct dm_target_msg dm_target_msg;
+                char msg_text[STRLEN("@cancel_deferred_remove") + 1];
+        } message = {
+                .dm_ioctl = {
+                        .version = {
+                                DM_VERSION_MAJOR,
+                                DM_VERSION_MINOR,
+                                DM_VERSION_PATCHLEVEL
+                        },
+                        .data_size = dm_size,
+                        .data_start = sizeof(struct dm_ioctl),
+                },
+                .dm_target_msg = {
+                        .sector = 0,
+                },
+        };
+
+        assert(name);
+
+        if (strlen(name) >= DM_NAME_LEN)
```

Comment:
`sizeof(message.dm_ioctl.name)` instead of `DM_NAME_LEN` would be nicer here, since the following strncpy_exact() uses that as size limit, and we verify the size here before writing after alll...